core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,438 @@
+          noteEdited = false // As no unsaved changes remain
+          enableDeleteNoteMenuItem()
+        } catch (e: IOException) {
+          e.printStackTrace()
+            .also { context.toast(R.string.note_save_unsuccessful, Toast.LENGTH_LONG) }
+        }
+      } else {
+        context.toast(R.string.note_save_successful, Toast.LENGTH_LONG)
+        Log.d(TAG, "Required folder doesn't exist")
+      }
+    } else {
+      context.toast(R.string.note_save_error_storage_not_writable, Toast.LENGTH_LONG)
+    }
+  }
+
+  private fun deleteNote() {
+    val notesFolder = File(zimNotesDirectory)
+    val noteFile =
+      File(notesFolder.absolutePath, "$articleNoteFileName.txt")
+    val noteDeleted = noteFile.delete()
+    if (noteDeleted) {
+      add_note_edit_text.text.clear()
+      disableMenuItems()
+      context.toast(R.string.note_delete_successful, Toast.LENGTH_LONG)
+    } else {
+      context.toast(R.string.note_delete_unsuccessful, Toast.LENGTH_LONG)
+    }
+  }
+
+  /* String content of the note text file given at:
+   *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+   * is displayed in the EditText field (note content area)
+   */
+  private fun displayNote() {
+
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    if (noteFile.exists()) {
+      writeNote(noteFile)
+    }
+
+    // No action in case the note file for the currently open article doesn't exist
+  }
+
+  private fun writeNote(noteFile: File) {
+    noteFileExists = true
+    val contents = noteFile.readText()
+    add_note_edit_text.setText(contents) // Display the note content
+    add_note_edit_text.setSelection(add_note_edit_text.text.length - 1)
+    enableShareNoteMenuItem() // As note content exists which can be shared
+    enableDeleteNoteMenuItem()
+  }
+
+  private fun shareNote() {
+
+    /* The note text file corresponding to the currently open article, given at:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * is shared via an app-chooser intent
+     * */
+    if (noteEdited) {
+      // Save edited note before sharing the text file
+      saveNote(add_note_edit_text.text.toString())
+    }
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    if (noteFile.exists()) {
+      val noteFileUri = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        // From Nougat 7 (API 24) access to files is shared temporarily with other apps
+        // Need to use FileProvider for the same
+        FileProvider.getUriForFile(
+          requireContext(), requireContext().packageName + ".fileprovider",
+          noteFile
+        )
+      } else {
+        Uri.fromFile(noteFile)
+      }
+      sendNoteFileShareIntent(noteFileUri)
+    } else {
+      context.toast(R.string.note_share_error_file_missing, Toast.LENGTH_SHORT)
+    }
+  }
+
+  private fun sendNoteFileShareIntent(noteFileUri: Uri?) {
+    val noteFileShareIntent = Intent(Intent.ACTION_SEND)
```

Comment:
this isn't a [scope function](https://kotlinlang.org/docs/reference/scope-functions.html), just a regular function, ayush meant for you to use `.apply` on the intent object